plugins/org.eclipse.osee.disposition.rest/src/org/eclipse/osee/disposition/rest/internal/DispoApiImpl.java

```diff
@@ -1076,18 +1065,65 @@ private List<DispoItem> massDisposition(BranchId branch, String setId, List<Stri
+                  }
                   annotation.setIsResolutionValid(true);
                   annotation.setIsConnected(true);
                   annotation.setIsDefault(false);
                   dispoConnector.connectAnnotation(annotation, item.getDiscrepanciesList());
                   newAnnotations.set(annotation.getIndex(), annotation);
+
+                  if (annotation.getIsPairAnnotation()) {
+                     DispoAnnotationData parentAnnotation = getParentAnnotation(branch, item.getGuid(), annotation);
+                     if (!parentAnnotation.getIsResolutionValid()) {
+                        String newParentResolutionType = resolutionType;
+                        String newParentResolution = resolution;
+
+                        List<String> satisfiedPairs = new ArrayList<>();
+                        for (int pairNumber : annotation.getPairedWith()) {
+                           String pairLoc = String.format("%s.%d", parentAnnotation.getLocationRefs(), pairNumber);
+                           DispoAnnotationData pairAnnotation = DispoUtil.getByLocation(newAnnotations, pairLoc);
+                           if (pairAnnotation.getIsResolutionValid()) {
+                              if (annotation.getRow() < pairNumber) {
+                                 satisfiedPairs.add(String.format("%s/%s", annotation.getRow(), pairNumber));
+                              } else {
+                                 satisfiedPairs.add(String.format("%s/%s", pairNumber, annotation.getRow()));
+                              }
+                              if (!pairAnnotation.getResolutionType().equals(newParentResolutionType)) {
+                                 newParentResolutionType =
+                                    newParentResolutionType + "/" + pairAnnotation.getResolutionType();
+                              }
+                              if (!pairAnnotation.getResolution().equals(newParentResolution)) {
+                                 newParentResolution = newParentResolution + "/" + pairAnnotation.getResolution();
+                              }
+                           }
+                        }
+                        if (!satisfiedPairs.isEmpty()) {
+                           DispoAnnotationData newParentAnnotation = new DispoAnnotationData();
+                           newParentAnnotation = parentAnnotation;
+                           newParentAnnotation.setResolutionType(newParentResolutionType);
+                           newParentAnnotation.setResolution(newParentResolution);
+                           newParentAnnotation.setSatisfiedPairs(
+                              String.format("Pairs Satisfied By: %s", satisfiedPairs.toString()));
+                           if (newParentAnnotation.getIsAnalyze()) {
+                              newParentAnnotation.setIsAnalyze(true);
+                           }
+                           newParentAnnotation.setIsResolutionValid(true);
```

Comment:
Instead of doing 'set' everywhere in this file, can we just pass in these values via constructor?

Reply:
This specific case, we have the newParentAnnotation is initially set based off of a previously existing Annotation.
We only want to change specifically the referenced values.